src/test/java/test/github1490/VerifyDataProviderListener.java

```diff
@@ -0,0 +1,113 @@
+package test.github1490;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.testng.Assert;
+import org.testng.IDataProviderMethod;
+import org.testng.TestNG;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.Test;
+import org.testng.xml.XmlSuite;
+import org.testng.xml.XmlTest;
+import test.SimpleBaseTest;
+import test.listeners.github1490.DataProviderInfoProvider;
+import test.listeners.github1490.LocalDataProviderListener;
+
+import java.util.Arrays;
+
+public class VerifyDataProviderListener extends SimpleBaseTest {
```

Comment:
Could you add tests where:
* 2 or more methods are sharing the same data provider?
* 2 or more factories are sharing the same data provider?
* 1 or more methods and 1 or more factories are sharing the same data provider?

Reply:
I can do that, but what are we going to be asserting here ? The functionality of a data provider is already a well established part. We are only adding a listener capability to it. So I would be curious to know, as to what is supposed to be the outcome of the tests.

Reply:
The goal is to cover the feature as much as possible.
It will check if the method is really called with many methods, only once, and maybe in the same order (even if the order is not specified/guaranteed).
It is the common issues we already had with current listeners.